_test/test_utilities/test_warnonce.m

```diff
@@ -0,0 +1,29 @@
+classdef test_warnonce < TestCase
+
+    methods
+        function obj = test_warnonce(~)
+            obj@TestCase('test_warnonce');
+        end
+
+        function test_warnonce_warns_once_defaults(~)
+            out = evalc("warnonce('HORACE:panic:panic', 'Panic at the disco')");
+            msg = lastwarn;
+
+            assertEqual(msg, 'Panic at the disco');
+            assertTrue(contains(out, 'Panic at the disco'))
```

Comment:
The test will fail if run on the same machine twice. 

This is pain for maintenance. 
Reset mechanism should be added for testing purposes.

Reply:
Double check you're happy with the new mechanism and we can start putting this in places where it's used.